ts/src/coinlist.ts

```diff
@@ -0,0 +1,2246 @@
+            const minWithdrawal = this.safeString (currency, 'min_withdrawal', undefined);
+            result[code] = {
+                'id': id,
+                'code': code,
+                'name': code,
+                'info': currency,
+                'active': active,
+                'deposit': depositEnabled,
+                'withdraw': withdrawEnabled,
+                'fee': undefined,
+                'precision': precision,
+                'limits': {
+                    'amount': { 'min': undefined, 'max': undefined },
+                    'withdraw': { 'min': minWithdrawal, 'max': undefined },
+                },
+                'networks': {},
+            };
+        }
+        return result;
+    }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name coinlist#fetchMarkets
+         * @description retrieves data on all markets for coinlist
+         * @param {object} [params] extra parameters specific to the exchange api endpoint
+         * @returns {object[]} an array of objects representing market data
+         */
+        const response = await this.publicGetV1Symbols (params);
+        //
+        //     {
+        //         symbols: [
+        //             {
+        //                 symbol: 'CQT-USDT',
+        //                 base_currency: 'CQT',
+        //                 is_trader_geofenced: false,
+        //                 list_time: '2021-06-15T00:00:00.000Z',
+        //                 type: 'spot',
+        //                 series_code: 'CQT-USDT-SPOT',
+        //                 long_name: 'Covalent',
+        //                 asset_class: 'CRYPTO',
+        //                 minimum_price_increment: '0.0001',
+        //                 minimum_size_increment: '0.0001',
+        //                 quote_currency: 'USDT',
+        //                 index_code: null,
+        //                 price_band_threshold_market: '0.05',
+        //                 price_band_threshold_limit: '0.25',
+        //                 last_price: '0.12160000',
+        //                 fair_price: '0.12300000',
+        //                 index_price: null
+        //             },
+        //         ]
+        //     }
+        //
+        const markets = this.safeValue (response, 'symbols');
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = this.safeString (market, 'symbol');
+            const baseId = this.safeString (market, 'base_currency');
+            const quoteId = this.safeString (market, 'quote_currency');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const amountPrecision = this.safeString (market, 'minimum_size_increment');
+            const pricePrecision = this.safeString (market, 'minimum_price_increment');
```

Comment:
if we're not passing the values through a parser that will convert them to the appropriate type, we should use `safeNumber` to parse numeric values